tests/plugins/kernel_install/grub_test.sh

```diff
@@ -17,11 +17,22 @@ function test_grub()
   local expected_cmd
 
   output=$(run_bootloader_update 'TEST_MODE' 'local')
-  expected_cmd='sudo -E grub-mkconfig -o /boot/grub/grub.cfg'
+
+  if [[ ! -x "$(command -v "grub-mkconfig")" ]]; then
+    expected_cmd='sudo -E grub2-mkconfig -o /boot/grub2/grub.cfg'
+  else
+    expected_cmd='sudo -E grub-mkconfig -o /boot/grub/grub.cfg'
+  fi
```

Comment:
Correct me if I'm wrong, but you assume that the developer system has grub, right? I guess this test will fail for the simple fact that the developer is using another bootloader such as u-boot or syslinux, right? I think you can mock some of the grub behavior with an alias in the test; that way, you can also test the `grub` and `grub2`. IIRC `device_test` and `utils_test.sh` might have some good examples.

Reply:
Currently, the test fails if the developer is using another bootloader. For example, all tests run fine on the CI/CD, but, locally with Fedora, tests are failing. I'll try to mock the grub behavior so that the tests can run system wide.

Reply:
Checking the `device_test` and `utils_test.sh`, I saw how mock was done on unit tests. But, I'm still running into a problem: even if I alias or export grub-mkconfig, `$(command -v 'grub-mkconfig') cannot find the mock command and I end up falling into the case where there is no grub installed.

Have you got any suggestions on what to do?

Reply:
End up getting a solution for the `$(command -v 'grub-mkconfig')` problem with `kwlib`.